dace/codegen/common.py

```diff
@@ -94,3 +97,50 @@ def unparse_interstate_edge(code_ast: Union[ast.AST, str], sdfg: SDFG, symbols=N
     strio = StringIO()
     InterstateEdgeUnparser(sdfg, code_ast, strio, symbols, codegen)
     return strio.getvalue().strip()
+
+
+@lru_cache()
+def get_gpu_backend() -> str:
+    """
+    Returns the currently-selected GPU backend. If automatic,
+    will perform a series of checks to see if an NVIDIA device exists,
+    then if an AMD device exists, or fail.
+    Otherwise, chooses the configured backend in ``compiler.cuda.backend``.
+    """
+    backend: str = config.Config.get('compiler', 'cuda', 'backend')
```

Comment:
Does it make sense to keep calling this `cuda` instead of `gpu`? I suppose that the CUDA code is "hipified," and that warrants keeping the name, but maybe it should be changed in the future.

Reply:
We should, it’s a bigger refactor PR though